app/services/datetime_parser.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class DatetimeParser
+  def initialize(date:, time:)
+    @raw_date = date.to_h
+    @raw_time = time.to_s
+  end
+
+  def parse
+    @issue_items = []
+
+    check_date_is_valid
+    check_time_is_valid
+    return if issue_items.any?
```

Comment:
Should this raise an exception if there are any issues?

Reply:
At this point in time, `issue_items` isn't really doing anything. Perhaps it would be better to extract all of the error handling code into a separate commit?